mail/views.py

```diff
@@ -0,0 +1,44 @@
+"""
+Views for email REST APIs
+"""
+import logging
+
+from rest_framework import (
+    authentication,
+    permissions,
+)
+from rest_framework.views import APIView
+from rest_framework.response import Response
+
+from search.api import (
+    prepare_and_execute_search,
+    get_all_query_matching_emails
+)
+from mail.api import MailgunClient
+from mail.permissions import UserCanMessageLearnersPermission
+
+
+log = logging.getLogger(__name__)
+
+
+class MailView(APIView):
+    """
+    View class that authenticates and handles HTTP requests to mail API URLs
+    """
+    authentication_classes = (authentication.SessionAuthentication, )
+    permission_classes = (permissions.IsAuthenticated, UserCanMessageLearnersPermission, )
+
+    def post(self, request, *args, **kargs):  # pylint: disable=unused-argument, no-self-use, missing-docstring
+        emails = prepare_and_execute_search(
+            request.user,
+            search_param_dict=request.data['search_request'],
+            search_func=get_all_query_matching_emails
+        )
+        mailgun_resp = MailgunClient.send_bcc(
```

Comment:
for our purposes, i haven't seen anything compelling. we haven't looked into or decided what we want in terms of alerting the user when they send an email. if we want to do some specific notifications, we can look deeper into the mailgun response. right now we dont tell the user _anything_ about their attempted send